x-pack/plugins/enterprise_search/server/lib/indices/delete_access_control_index.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core/server';
+
+import { CONNECTORS_ACCESS_CONTROL_INDEX_PREFIX } from '../..';
+
+export const deleteAccessControlIndex = async (client: IScopedClusterClient, index: string) => {
+  const accessControlExists = await client.asCurrentUser.indices.exists({
```

Comment:
suggestion: did you consider a try/catch here to avoid an extra call to Elasticsearch?

Reply:
No. It is a good idea though.